### DISCUSSION TITLE
How to connect multiple USB devices (keyboard and mouse, in my case) to a Pico

### DISCUSSION THREAD

**Discussion by horsepork** [sekigon-gonnoc/Pico-PIO-USB#108] [Q&A]
I have managed to get a keyboard and mouse to send their HID reports separately, but I am not sure what the steps are to get multiple devices to connect. I attempted to use a USB hub, but it was behaving oddly and causing the code to occasionally hang. I also attempted to make two instances of the TinyUSB host, each using different pairs of pins for the PIO. However, whichever host begins first is the one that will register devices being (un)mounted or receive HID reports. Any pointers? And to clarify, I do not need the Pico to function as USB device, either over the main USB port or a PIO-based port.

**Comment by ghost** [comment]
Hi @horsepork
i answered your question here:
https://github.com/sekigon-gonnoc/Pico-PIO-USB/discussions/72
